pandas/tests/io/test_sql.py

```diff
@@ -1374,6 +1390,13 @@ def test_bigint(self):
 
         tm.assert_frame_equal(df, result)
 
+    def test_reading_ints(self):
+        # BigInt should be read as int64, Int should be read as int32, and SmallInt should be read as int16
+        df = sql.read_sql_query("SELECT * FROM types_test_data", self.conn)
+        assert issubclass(df.BigIntCol.dtype.type, np.int64)
```

Comment:
Rather than using is subclass I think can use equality to make sure we get the proper type

Reply:
@WillAyd this change has been incorporated